src/main/kotlin/frc/robot/subsystems/ExampleSubsystem.kt

```diff
@@ -0,0 +1,117 @@
+): SubsystemBase() {
+    init {
+        // This is run when the ExampleSubsystem instance is created.
+    }
+
+    /**
+     * Example command factory method.
+     *
+     * @return a command
+     */
+    fun exampleMethodCommand(): Command =
+        // Inline construction of command goes here.
+        // Subsystem::runOnce implicitly requires `this` subsystem.
+        runOnce {
+            println("This is an example command.")
+        }
+
+    /**
+     * An example of a regular kotlin property
+     * whose value cannot be changed.
+     */
+    private val digitalInput = DigitalInput(digitalInputID)
+    private val motor = PWMSparkMax(motorID)
+
+    /**
+     * An example of a regular kotlin property
+     * that can be modified.
+     */
+    private var exampleMutableProperty = 2.0
+
+    /**
+     * An example of a property with a custom getter.
+     * Every time this property is accessed, its value will be re-computed to a new value.
+     *
+     * Properties like these can be used to query a boolean state of the subsystem (for example, a digital sensor).
+     */
+    val exampleCondition: Boolean get() = motor.isAlive
+
+    /**
+     * An example of a property with a custom getter and a custom setter.
+     *
+     * When this property is assigned a value, the code within the set(){ }
+     * block will also run.
+     *
+     * Note that this property does not store any data,
+     * and is essentially a proxy for 2 different methods.
+     */
+    var exampleGetSetProperty: Double
+        get() = motor.get()
+        set(value) {
+            if (digitalInput.get()) motor.set(value)
+        }
+
+    /**
+     * An example of a getter/setter property with a backing field.
+     *
+     * A backing field acts like an invisible variable tied to the property itself,
+     * which can be accessed in the get() and set() methods of that property.
+     *
+     * In order to use a backing field, an initial value must be given to the property
+     * itself.
+     *
+     * To refer to a backing field, use the keyword "field".
+     *
+     * Properties with a backing field can exclude either the getter or the setter.
+     */
+    var exampleBackingFieldProperty = exampleConstructorArg
+        get() {
+            field++
+            return field
+        }
+        set(value) {
+            motor.stopMotor()
+            field = value + 1
+        }
+
+    /**
+     * An example subsystem property using a lazy initializer.
+     *
+     * A lazy initializer will initialize the property the first
+     * time it is accessed, rather than when the subsystem is created.
+     *
+     * A lazy initializer can be used to avoid expensive operations or
+     * resource allocation if an object is not used. For instance,
+     * giving a simulation object a lazy initializer can prevent it
+     * from using unnecessary CPU or memory on the real robot.
+     */
+    val lazyProperty: Int by lazy { digitalInput.channel }
```

Comment:
I don't think we need all of these. We could spend forever documenting every kotlin feature. I think this should show more FRC examples, rather than kotlin examples. Similar to the wpilib examples